tools/plugingenerator/plugin_generator.py

```diff
@@ -0,0 +1,229 @@
+    try:
+        os.mkdir(resPath, mode)
+        print("res directory created")
+    except FileExistsError:
+        print("'res' directory already exists!")
+    if generatorOptions["resources"]["resources_qrc"]:
+        resQrc = os.path.join(resPath, "resources.qrc")
+        if not os.path.exists(resQrc):
+            resFile = open(resQrc, "w")
+            resFile.write("<!DOCTYPE RCC>\n")
+            resFile.write('<RCC version="1.0"/>')
+            resFile.close()
+
+##################################################### Plugin CMakeLists #########################################
+if generatorOptions["plugin"]["cmakelists"]:
+    cmakeListsPath = os.path.join(newPluginPath, "CMakeLists.txt")
+    cmakeTemplate = Template(filename="templates/cmakelists_template.mako")
+    cmakeContent = cmakeTemplate.render(
+        scopy_module=pluginName, config=generatorOptions["cmakelists"]
+    )
+
+if not os.path.exists(cmakeListsPath):
+    cmakeListsFile = open(cmakeListsPath, "w")
+    cmakeListsFile.write(cmakeContent)
+    cmakeListsFile.close()
+    print("plugin CMakeLists created")
+else:
+    print("CMakeLists.txt already exists!")
+
+##################################################### Plugin ToolList #########################################
+toolList = generatorOptions["plugin"]["tools"]
+if toolList:
+    for t in toolList:
+        if os.path.exists(pluginIncludePath) and os.path.exists(srcPath):
+            includeFilePath = os.path.join(pluginIncludePath, t["file_name"] + ".h")
+            srcFilePath = os.path.join(srcPath, t["file_name"] + ".cpp")
+
+            includeFileTemplate = Template(
+                filename="templates/tool_header_template.mako"
+            )
+            includeFileContent = includeFileTemplate.render(
+                plugin_export_macro=pluginExportMacro, plugin_name=pluginName, config=t
+            )
+            includeFile = open(includeFilePath, "w")
+            if os.path.getsize(includeFile.name) == 0:
+                includeFile.write(includeFileContent)
+            includeFile.close()
+
+            srcFileTemplate = Template(filename="templates/tool_src_template.mako")
+            srcFileContent = srcFileTemplate.render(config=t)
+            srcFile = open(srcFilePath, "w")
+            if os.path.getsize(srcFile.name) == 0:
+                srcFile.write(srcFileContent)
+            srcFile.close()
+
+################################################### Plugin Src #################################################
+if os.path.exists(srcPath):
+    pluginSourceFilePath = os.path.join(srcPath, pluginName + ".cpp")
+    if not os.path.exists(pluginSourceFilePath):
+        pluginSrcTemplate = Template(filename="templates/plugin_src_template.mako")
+        pluginSrcContent = pluginSrcTemplate.render(config=generatorOptions["plugin"])
+        pluginSourceFile = open(pluginSourceFilePath, "w")
+        pluginSourceFile.write(pluginSrcContent)
+        pluginSourceFile.close()
```

Comment:
When finished, specify what needs to be added to plugins/CMakelists.txt to enable building the plugin